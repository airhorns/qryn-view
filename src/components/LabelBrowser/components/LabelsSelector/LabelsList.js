import { useMemo } from "react";

function LabelItem(props) {
    const { selected, label } = props;
    const isSelected = useMemo(() => selected, [selected]);

    const selectedStyle = useMemo(() => {
        if (isSelected)
            return {
                borderColor: "#11abab",
                color: "#11abab",
            };
        else return {};
    }, [isSelected]);
    return (
        <small style={selectedStyle} onClick={(e) => props.onClick(label)}>
            {label}
        </small>
    );
}

export default function LabelsList(props) {

    const { labels } = props;

    const onClick = (e) => {
        props.onLabelSelected(e);
    };

    const lsList = useMemo(() => labels, [labels]);

    return (
        <div className="valuelist-content">
            {lsList &&
                lsList.map((label, key) => (
                    <LabelItem
                        key={key}
                        label={label.name}
                        selected={label.selected}
                        onClick={onClick}
                    />
                ))}
        </div>
    );
}
