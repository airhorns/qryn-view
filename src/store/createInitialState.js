import * as moment from "moment";
import { environment } from "../environment/env.dev";
import stateFromQueryParams from "../helpers/stateFromQueryParams";
import localService from "../services/localService";
import localUrl from "../services/localUrl";

const debugLocal = () => {
    let isDebug = JSON.parse(localStorage.getItem("isDebug"));
    if (!isDebug) {
        localStorage.setItem("isDebug", JSON.stringify({ isActive: false }));
        isDebug = {isActive:false}
    } 

    return isDebug;
};

export default function initialState() {
    const urlState = stateFromQueryParams();
    const historyService = localService().historyStore();
    const linkService = localUrl();
    const lastQuery = linkService.getAll()?.[0]?.params

    const state = {
        debugMode: debugLocal().isActive|| false,
        labels: [],
        labelValues: [],
        queryHistory: historyService.getAll() || [],
        linksHistory: linkService.getAll() || [],
        timeRange: [],
        query: urlState.query || "",
        logs: [],
        matrixData: [],
        loading: false,
        start:
            urlState.start ||
            new Date(
                moment(Date.now())
                    .subtract(5, "minutes")
                    .format("YYYY-MM-DDTHH:mm:ss.SSSZ")
            ),
        stop:
            urlState.end ||
            new Date(moment(Date.now()).format("YYYY-MM-DDTHH:mm:ss.SSSZ")),
        from: urlState.from || null,
        to: urlState.to || null,
        label: urlState.label || "Last 5 minutes",
        messages: [],
        limitLoad: false,
        limit: urlState.limit || 100,
        step: urlState.step || 100,
        rangeOpen: false,
        labelsBrowserOpen: false,
        settingsMenuOpen: false,
        timePickerOpen: false,
        settingsDialogOpen: false,
        historyOpen: false,
        apiErrors: "",
        urlQueryParams: urlState || {},
        urlLocation: "",
        apiUrl: urlState.apiUrl ||  environment.apiUrl || "",
        apiWarning:{},
        isSubmit: urlState.isSubmit || false,
        isEmbed: urlState.isEmbed || false,
        chartType: "line",
        notifications: [],
        theme: "darkTheme",
    };
    const debug = state.debugMode;
    if (debug) console.log("ðŸš§ LOGIC/ INITIAL STATE ::: ", state);

    return state;
}
