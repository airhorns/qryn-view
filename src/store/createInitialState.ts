import moment, { Moment, MomentInput } from "moment";
import { nanoid } from "nanoid";
import { environment } from "../environment/env.dev";
import stateFromQueryParams from "../helpers/stateFromQueryParams";
import localService from "../services/localService";
import localUrl from "../services/localUrl";

export interface IsDebug {
    isActive: boolean;
}

const debugLocal = (): boolean => {
    const debugFromLocal: IsDebug = (JSON.parse(
        localStorage.getItem("isDebug") || JSON.stringify(false)
    ) as IsDebug) || { isActive: false };

    const isDebugItem: boolean = debugFromLocal.isActive === true;

    if (!isDebugItem) {
        localStorage.setItem("isDebug", JSON.stringify({ isActive: false }));
        return false;
    }

    return true;
};

export interface URLState {
    queryType: string;
    start: string | Date;
    time: string;
    to: string | null;
    stop: string | Date;
    end: string | Date;
    from: string | Date;
    left: any[];
    right: any[];
    label: string;
    limit: string | number;
    step: string | number;
    apiUrl: string;
    isSubmit: boolean;
    isEmbed: boolean;
    theme: string;
    isSplit: boolean;
}

export const initialUrlState = {
    queryType: "range",
    start: "",
    time: "",
    to: "",
    stop: "",
    end: "",
    from: "",
    left: [],
    right: [],
    label: "",
    limit: 100,
    step: 100,
    apiUrl: "",
    isSubmit: false,
    isEmbed: false,
    theme: "",
    isSplit: false,
};

export function date_fm(input: MomentInput): Moment {
    return moment(input);
}

export default function initialState() {
    const urlState: URLState = stateFromQueryParams() || initialUrlState;
    const historyService = localService().historyStore();
    const linkService = localUrl();
    const state = {
        debugMode: debugLocal() || false,
        labels: [],
        labelValues: [],
        queryHistory: historyService.getAll() || [],
        linksHistory: linkService.getAll() || [],
        timeRange: [],
        queryType: urlState.queryType || "range",
        logs: [],
        matrixData: [],
        vectorData: {},
        loading: false,
        queryResolution: 1,
        start:
            urlState.start ||
            new Date(
                moment(Date.now())
                    .subtract(5, "minutes")
                    .format("YYYY-MM-DDTHH:mm:ss.SSSZ")
            ),
        time: urlState.time || "", // for instant queries
        stop:
            urlState.end ||
            new Date(moment(Date.now()).format("YYYY-MM-DDTHH:mm:ss.SSSZ")),
        from: urlState.from || null,
        to: urlState.to || null,
        label: urlState.label || "Last 5 minutes",
        messages: [],
        limitLoad: false,
        limit: urlState.limit || 100,
        step: urlState.step || 100,
        rangeOpen: false,
        labelsBrowserOpen: true,
        settingsMenuOpen: false,
        timePickerOpen: false,
        settingsDialogOpen: false,
        historyOpen: false,
        apiErrors: "",
        urlQueryParams: urlState || {},
        urlLocation: "",
        apiUrl: urlState.apiUrl || environment.apiUrl || "",
        isSubmit: urlState.isSubmit || false,
        isEmbed: urlState.isEmbed || false,
        // dont mention queries // its obvious
        left: urlState['left']||[
            {
                id: nanoid(),
                idRef: "A",
                lastIdx: 1,
                panel: "left",
                queryType: "instant",
                limit: 100,
                step: 100,
                tableView: false,
                browserOpen: false,
                expr: "",
                labels: [], // name: selected:
                values: [], // label name selected
                //find query by ID and append response
            },
        ],

        right: urlState['right']||[
            {
                id: nanoid(),
                idRef: "A",
                lastIdx: 1,
                panel: "right",
                queryType: "instant",
                limit: 100,
                step: 100,
                tableView: false,
                browserOpen: false,
                expr: "",
                labels: [], // name: selected:
                values: [], // label name selected
                //find query by ID and append response
            },
        ],

        leftDataView: {
            type: "",
            data: [],
        },
        rightDataView: {
            type: "",
            data: [],
        },

        chartType: "line",
        resposeType: "",
        notifications: [],
        tableData: {},
        isTableView: false,
        theme: urlState.theme || "dark",
        isEmptyView: false,
        isSplit: false,
    };

    const debug = state.debugMode;
    if (debug) console.log("ðŸš§ LOGIC/ INITIAL STATE ::: ", state);

    return state;
}
