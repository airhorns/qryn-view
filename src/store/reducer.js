export default (state, action) => {
    switch (action.type) {
        case "SET_LABELS":
            return { ...state, labels: action.labels };
        case "SET_LOADING":
            return { ...state, loading: action.loading };
        case "SET_LOGS":
            return { ...state, logs: action.logs };
        case "SET_LABEL_VALUES":
            return {...state, labelValues: action.labelValues};
        default:
            return { ...state };
    }
};