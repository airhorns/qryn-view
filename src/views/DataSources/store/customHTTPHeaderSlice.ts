import { createSlice } from "@reduxjs/toolkit";
import { nanoid } from "nanoid";

export interface CustomHttpHeader {
    id: string;
    header: string;
    value: string;
}

export interface CustomHttpHeaderState {
    customHttpHeaders: CustomHttpHeader[];
}

const initialState: CustomHttpHeaderState = {
    customHttpHeaders: [
        {
            id: nanoid(),
            header: "",
            value: "",
        },
    ],
};

export interface CustomHttpHeaderPayload {
    payload: CustomHttpHeader;
}

export interface CustomHttpPayload {
    payload: CustomHttpHeaderState;
}

const addCustomHttpHeader = (
    state: CustomHttpHeaderState,
    action: CustomHttpHeaderPayload
) => {
    const { header, value } = action.payload;
    state.customHttpHeaders = [
        ...state.customHttpHeaders,
        { id: nanoid(), header: header, value: value },
    ];
};

const removeCustomHttpHeader = (
    state:CustomHttpHeaderState,
    action: CustomHttpHeaderPayload
) => {
    const {id} = action.payload
    const filtered = state.customHttpHeaders.filter(f => f.id !== id)
    state.customHttpHeaders = filtered
}

const updateCustomHttpHeader = (
    state:CustomHttpHeaderState,
    action: CustomHttpHeaderPayload
) => {
    const {id} = action.payload
    const stateClone = JSON.parse(JSON.stringify(state.customHttpHeaders))
    stateClone.forEach( (header:CustomHttpHeader )=>{
        if(header.id === id) {
            header = action.payload
        }
    })
    state.customHttpHeaders = stateClone
} 

export const customHTTPHeaderSlice = createSlice({
    name: "customHTTPHeader",
    initialState,
    reducers: {
        addCustomHttpHeader,
        removeCustomHttpHeader,
        updateCustomHttpHeader

    },
});
