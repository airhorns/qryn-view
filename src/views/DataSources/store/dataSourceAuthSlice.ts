import {createSlice} from '@reduxjs/toolkit'

export interface DataSourceAuthState {
    basicAuth: boolean;
    withCredentials: boolean;
    TLSClientAuth: boolean;
    withCACert: boolean;
    skipTLSVerify: boolean;
    forwardOAuthIdentity: boolean;
}

const initialState:DataSourceAuthState = {
    basicAuth: false,
    withCredentials: false,
    TLSClientAuth: false,
    withCACert: false,
    skipTLSVerify: false,
    forwardOAuthIdentity: false,
}

export interface BasicAuthPayload  {
  payload:boolean;
}
export interface WithCredentialsPayload  {
    payload:boolean;
  }
  export interface TLSCLientAuthPayload  {
    payload:boolean;
  }
  export interface WithCACertPayload  {
    payload:boolean;
  }
  export interface SkipTLSVerifyPayload  {
    payload:boolean;
  }
  export interface ForwardOauthIdentityPayload  {
    payload:boolean;
  }
  

const setBasicAuth = (state:DataSourceAuthState, action:BasicAuthPayload) => {
    state.basicAuth = action.payload;
}

const setWithCredentials = (state:DataSourceAuthState, action:WithCredentialsPayload) => {
    state.withCredentials = action.payload;
}

const setTLSClientAuth = (state:DataSourceAuthState, action:TLSCLientAuthPayload)=>{
state.TLSClientAuth = action.payload;
}

const setWithCACert = (state: DataSourceAuthState, action: WithCACertPayload) => {
    state.withCACert = action.payload;
}

const setSkipTLSVerify = (state:DataSourceAuthState, action: SkipTLSVerifyPayload) => {
    state.skipTLSVerify = action.payload;
}

const setForwardOauthIdentity = (state:DataSourceAuthState, action:ForwardOauthIdentityPayload) => {
    state.forwardOAuthIdentity = action.payload;
}


export const dataSourceAuthSlice = createSlice({
    name:'dataSourceAuth',
    initialState,
    reducers: {
        setBasicAuth,
        setWithCredentials,
        setTLSClientAuth,
        setWithCACert,
        setSkipTLSVerify,
        setForwardOauthIdentity

    },  
});

