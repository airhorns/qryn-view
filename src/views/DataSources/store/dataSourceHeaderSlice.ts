import {createSlice} from '@reduxjs/toolkit'

export interface DataSourceHeaderState {
    name: string;
    isDefault: boolean;
}

const initialState:DataSourceHeaderState = {
    name: 'Logs',
    isDefault:true,
}
export interface NamePayload  {
  payload:string
}

export interface IsDefaultPayload {
    payload: boolean;
}

const setName = (state:DataSourceHeaderState, action:NamePayload) => {
    state.name = action.payload
}

const setIsDefault = (state:DataSourceHeaderState, action: IsDefaultPayload) => {
    state.isDefault = action.payload
}
export const dataSourceHeaderSlice = createSlice({
    name:'dataSourceHeader',
    initialState,
    reducers: {
        setName,
        setIsDefault

    },  
})