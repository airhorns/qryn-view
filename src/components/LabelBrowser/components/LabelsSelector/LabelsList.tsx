import { useMemo } from "react";
import styled from "@emotion/styled";
import ShowLogsButton from "../Buttons/ShowLogsButton";
import { queryBuilder } from "../../helpers/querybuilder";
import { useDispatch, useSelector } from "react-redux";
import { setLeftPanel } from "../../../../actions/setLeftPanel";
import { setRightPanel } from "../../../../actions/setRightPanel";
import store from "../../../../store/store";

function panelAction(name: any, value: any) {
    if (name === "left") {
        return setLeftPanel(value);
    }
    return setRightPanel(value);
}
export const EmptyLabels = (props: any) => {
    
    const EmptyCont = styled.div`
        color: ${({ theme }: any) => theme.textColor};
        padding: 10px;
    `;

    return <EmptyCont>No labels available, please adjust API</EmptyCont>;
};

// split labels for metrics and labels for logs


function LabelItem(props: any) {
    const { selected, label, type } = props;

    const isSelected = useMemo(() => selected, [selected]);

    const selectedStyle = useMemo(() => {
        if (isSelected)
            return {
                borderColor: "#11abab",
                color: "#11abab",
            };
        else return {};
    }, [isSelected]);
    return (
        <small
            className={type}
            style={selectedStyle}
            onClick={(e) => props.onClick(label)}
        >
            {label}
        </small>
    );
}

export default function LabelsList(props: any) {
    const dispatch = useDispatch();
    const { labels, data, name } = props;
    const { dataSourceType } = data;
    const panelQuery = useSelector((store: any) => store[name]);

    const onClick = (e: any) => {
        if (e === "Select Metric") {
            props.onLabelSelected("__name__");
        }
        props.onLabelSelected(e);
    };

    const onMetricOptionsClick = (e: any) => {

        console.log(e)
    }

    const lsList = useMemo(() => {
        if (dataSourceType !== "metrics") {
            return labels;
        }

        return labels?.filter((f: any) => f.name !== "__name__");
    }, [dataSourceType, labels]);

    const useQuery = () => {
        const qu = queryBuilder(data.labels, data.expr)
        const panel = [...panelQuery];
        panel.forEach((query) => {
            if (query.id === props.data.id) {
                query.expr = qu;
                query.labels = data.labels
            }
        });
        dispatch(panelAction(name, panel));
    }
    return (
        <div className="valuelist-content">
            {/* {metricLabel !== null && lsList && dataSourceType === 'metrics' && (
                <LabelItem
                    type={"metric"}
                 
                    label={"Select Metric"}
                    selected={metricLabel?.selected}
                    onClick={onMetricOptionsClick}
                 //   onClick={onClick}
                />
            )} */}

            {lsList &&
                <>
                    {lsList.map((label: any, key: any) => (
                        <LabelItem
                            key={key}
                            label={label.name}
                            selected={label.selected}
                            onClick={onClick}
                        />
                    ))}
                    <ShowLogsButton
                        onClick={useQuery}
                        isMobile={false}
                        alterText={"Use Query"}
                        loading={false}
                    />
                </>
            }
                
            {!lsList && <EmptyLabels {...props} />}
        </div>
    );
}
