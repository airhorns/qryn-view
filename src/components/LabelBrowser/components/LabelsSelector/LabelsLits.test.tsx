import '@testing-library/jest-dom'
import {cleanup} from '@testing-library/react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'
import { Provider } from 'react-redux';
import LabelsList from './LabelsList';
import store from '../../../../store/store'
import { mockPropsData } from '../../../../../mocks/baseProps.mock';
Enzyme.configure({adapter: new Adapter()});
// // Note: running cleanup afterEach is done automatically for you in @testing-library/react@9.0.0 or higher
// // unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

describe('Label List Component', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Provider store={store}><LabelsList {...mockPropsData}></LabelsList></Provider>)
        expect(toJson(wrapper.dive())).toMatchSnapshot();
      });
});
