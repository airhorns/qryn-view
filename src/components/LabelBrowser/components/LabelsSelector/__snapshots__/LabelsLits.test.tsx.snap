// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label List Component renders correctly 1`] = `
<LabelsList
  data={
    Object {
      "browserOpen": true,
      "chartView": false,
      "dataSourceId": "cHI2SqPzH_kxYRXj",
      "dataSourceType": "logs",
      "dataSourceURL": "https://view-oss-demo.qryn.dev",
      "expr": "",
      "id": "uLm84kZcbJXfDSEoU0NBp",
      "idRef": "L-A",
      "isShowTs": true,
      "labels": Array [
        Object {
          "name": "code",
          "selected": true,
          "values": Array [
            Object {
              "id": "I8JWIMUkRRq5jvDfE4r0l",
              "inverted": false,
              "label": "code",
              "name": "503",
              "selected": true,
              "type": "value",
            },
            Object {
              "id": "yszV3wcnGo-RHEMOXE5VI",
              "inverted": false,
              "label": "code",
              "name": "500",
              "selected": true,
              "type": "value",
            },
          ],
        },
      ],
      "lastIdx": 1,
      "limit": 100,
      "loading": false,
      "panel": "left",
      "queryType": "range",
      "response": Object {},
      "step": 5,
      "tableView": false,
      "values": Array [],
    }
  }
  name="left"
  queries={
    Array [
      Object {
        "browserOpen": true,
        "chartView": false,
        "dataSourceId": "cHI2SqPzH_kxYRXj",
        "dataSourceType": "logs",
        "dataSourceURL": "https://view-oss-demo.qryn.dev",
        "expr": "",
        "id": "uLm84kZcbJXfDSEoU0NBp",
        "idRef": "L-A",
        "isShowTs": true,
        "labels": Array [
          Object {
            "name": "code",
            "selected": true,
            "values": Array [
              Object {
                "id": "I8JWIMUkRRq5jvDfE4r0l",
                "inverted": false,
                "label": "code",
                "name": "503",
                "selected": true,
                "type": "value",
              },
              Object {
                "id": "yszV3wcnGo-RHEMOXE5VI",
                "inverted": false,
                "label": "code",
                "name": "500",
                "selected": true,
                "type": "value",
              },
            ],
          },
        ],
        "lastIdx": 1,
        "limit": 100,
        "loading": false,
        "panel": "left",
        "queryType": "range",
        "response": Object {},
        "step": 5,
        "tableView": false,
        "values": Array [],
      },
    ]
  }
  width={1256}
/>
`;
